### PR TITLE
Fix the probability weight for ND and 1D distributions

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------

This a quick fix, and this bug also exists in the devel branch. When we do the restructure for ND distribution, the bug will finally be fixed in the devel branch. 

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by alfoa** [comment]
@wangcj05 do you want me to merge this or we wait for a final PR?

**Comment by wangcj05** [comment]
Let's wait for the final PR.

**Comment by wangcj05** [comment]
Ran 472 tests in 631.4 seconds
370 passed, 60 skipped, 0 pending, 102 FAILED

**Comment by wangcj05** [comment]
There is no changes in the data object for this PR. But I do test all the unit tests, and they all pass. This is just a record.

**Comment by PaulTalbot-INL** [comment]
For the record, the same HDF5 error messages show up on the Mac test machine from the civet run of #483.  It shows 58 passing and 444 failed.  We need to address this before assuming we actually have 300+ passing.

**Comment by wangcj05** [comment]
(raven_libraries) [wangc][~/projects/wangc/raven]> python compare_tests.py reference_test_results.txt new_development_test_results.txt
**** SUMMARY ****
*********************************
**** THE FOLLOWING TESTS CHANGED OUTCOMES ****
**** Test framework.NDGridProbabilityWeightCDF | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework.test_push_into_hdf5 | has changed outcome from <OK> to <FAILED> !!!!
**** Test framework/PostProcessors/BasicStatistics.general | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/Metrics.multipleInputs | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/Metrics.simpleMetric | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/pca_postprocessor.timeDepImportanceRank | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/pca_postprocessor.transformationMatrix | has changed outcome from <FAILED> to <OK> !!!!
*****************
Number of success first file:
365
Number of success second file:
370
NUMBER OF SUCCESSES INCREASED: +5
TOTAL TIME OF TESTS DECREASED       : reference_time(650.2) > new_time(631.4)
AVERAGE TIMES FOR SUCCESS TESTS ARE : reference_time(1.78136986301) v.s. new_time(1.70648648649)
*********************************

**Comment by wangcj05** [comment]
After rebase:
(raven_libraries) [wangc][~/projects/wangc/raven]> python compare_tests.py reference_test_results.txt new_development_test_results.txt
**** SUMMARY ****
*********************************
**** THE FOLLOWING TESTS CHANGED OUTCOMES ****
**** Test framework.NDGridProbabilityWeightCDF | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/BasicStatistics.general | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/Metrics.multipleInputs | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/Metrics.simpleMetric | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/pca_postprocessor.timeDepImportanceRank | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/pca_postprocessor.transformationMatrix | has changed outcome from <FAILED> to <OK> !!!!
*****************
Number of success first file:
365
Number of success second file:
371
NUMBER OF SUCCESSES INCREASED: +6
TOTAL TIME OF TESTS DECREASED       : reference_time(650.2) > new_time(628.8)
AVERAGE TIMES FOR SUCCESS TESTS ARE : reference_time(1.78136986301) v.s. new_time(1.6948787062)

File: tests/framework/PostProcessors/BasicStatistics/test_BasicStatistics.xml [hunk 1]
```diff
@@ -294,30 +294,6 @@
...
-      <source>checkIfMatrixCanBeComputed_basicStatPP</source>
-    </Print>
-    <Print name="autoCorrelationTest_basicStatPP_dump">
-      <type>csv</type>
-      <source>autoCorrelationTest_basicStatPP</source>
-    </Print>
-    <Print name="CreateInfinity_basicStatPP_dump">
-      <type>csv</type>
-      <source>CreateInfinity_basicStatPP</source>
-    </Print>
```
[the remark this answers is not in the document]
Comment (wangcj05) [reply tests/framework/PostProcessors/BasicStatistics/test_BasicStatistics.xml]: These are duplicated in the test

File: framework/Samplers/Sampler.py [hunk 2]
```diff
@@ -326,6 +326,10 @@ def _readMoreXMLbase(self,xmlNode):
       for var in self.distributions2variablesMapping[distName]:
         position = utils.first(var.values())
         positionList.append(position)
+      if sum(list(set(positionList))) > 1 and len(positionList) != len(list(set(positionList))):
```
[the remark this answers is not in the document]
Comment (wangcj05) [reply framework/Samplers/Sampler.py]: fixed.

File: framework/Samplers/Sampler.py [hunk 3]
```diff
@@ -326,6 +326,10 @@ def _readMoreXMLbase(self,xmlNode):
...
+        dups = set(var for var in positionList if positionList.count(var) > 1)
```
[the remark this answers is not in the document]
Comment (wangcj05) [reply framework/Samplers/Sampler.py]: I think it should be a set.

File: tests/framework/PostProcessors/Metrics/gold/mcMetricRun/dumpMC.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<DataObjectMetadata name="outputDataMC">
+  <DataSet type="Static">
+    <general>
+      <inputs>x1,x2</inputs>
+      <outputs>ans,ans2</outputs>
+      <sampleTag>RAVEN_sample_ID</sampleTag>
+    </general>
+  </DataSet>
+  
+</DataObjectMetadata>
```
[the remark this answers is not in the document]
Comment (wangcj05) [reply tests/framework/PostProcessors/Metrics/gold/mcMetricRun/dumpMC.xml]: We can get rid of xml files.

File: tests/framework/pca_postprocessor/gold/transform/test_importantRank_Dataset_dump.xml [hunk 5]
```diff
@@ -1,25 +0,0 @@
...
-      <sensitivityIndex_z_x1>time</sensitivityIndex_z_x1>
-      <sensitivityIndex_x_x1>time</sensitivityIndex_x_x1>
-    </dims>
-    <general>
-      <outputs>importanceIndex_x_y1,importanceIndex_x_x1,importanceIndex_y_y1,importanceIndex_y_x1,importanceIndex_z_y1,importanceIndex_z_x1,sensitivityIndex_x_y1,sensitivityIndex_x_x1,sensitivityIndex_y_y1,sensitivityIndex_y_x1,sensitivityIndex_z_y1,sensitivityIndex_z_x1,pcaIndex_y1,pcaIndex_x1</outputs>
-      <sampleTag>RAVEN_sample_ID</sampleTag>
-    </general>
-  </DataSet>
-  
-</DataObjectMetadata>
```
Comment (PaulTalbot-INL) [inline tests/framework/pca_postprocessor/gold/transform/test_importantRank_Dataset_dump.xml]: Is this worth testing for its output?  Or is it tested somewhere else?

Comment (wangcj05) [reply tests/framework/pca_postprocessor/gold/transform/test_importantRank_Dataset_dump.xml]: I think the csv file should be enough.

**Comment by wangcj05** [comment]
@alfoa Can you merge this PR?

**Pull request merged**